### ISSUE TITLE
Sceptre init cmd

### ISSUE THREAD

**Issue by jamesroutley** [cloudreach/sceptre#112]
Add an `init` command, to quickly initialise new projects. 

The init command should be able to initialise:

- projects (new `config/` and `templates/` directories)
- environment (new directory with a `config.yaml`)
- stack (new `<name>.yaml`)

Files should contain example data, with optional parameters commented out:

```yaml
project_code: prj
region: us-east-1
# iam_role: aws:arn...
# template_bucket_name: my_sceptre_templates
...
```

**Comment by theseanything** [comment]
Be awesome if the stack init functionality could automatically populate the parameters section of a stack config from the cloudformation template.

**Comment by jamesroutley** [comment]
Wowow badass yeah let's do that

**Comment by raids** [comment]
To clarify, is this `sceptre init` to create a fresh directory structure with example stack and configs, or `sceptre init-stack`-type functionality to create a prepopulated config file from a pre-existing template?

I'd think they're both useful, but the latter might be more used. Happy to PR it and open a new issue for the former.

For `sceptre init`, how about:
- Check for `config` and `template` dirs, raise exception if one or the other already exists
- Create above dirs, `dev` dir in `config/` and an example stack (template and dev config)
- Finish by echoing what's happened and some commands to get started with the example stack

**Comment by theseanything** [comment]
Hey @raids , so I had already started development on this. The structure was going to be:
- sceptre init project <project_name>
- sceptre init env <env_path>
- sceptre init stack <stack_path> -t <template_path>
or I may condense it to 
- sceptre init project <project_name>
- sceptre init <env_path>|<stack_path>

Essentially init project will make a new folder with the project name with templates and config sub folders. It will then have an interactive prompt to pre-initialise a config.yaml file within the config folder. 

The env init function will create a new folder path within the config folder. It requires you to have already `cd` into an existing sceptre project. It will then have an interactive prompt to pre-initialise a config.yaml file within the environment. It will not include any values found in parent nested config.yamls and suggest suitable defaults. 

The stack init function will create a new stack config file. It should pre-initialise it with a template path and parse the template to pre-populate the parameters.

**Comment by raids** [comment]
Sweet, I think I actually prefer that.

Regarding `init` creating a new (root) folder with the project name, I won't have Sceptre installed until I already have that project directory already created, and a virtualenv within with the installation there. I assume this is a common pattern for many people, so is it worth considering doing everything after  the root project directory creation, just in the current working directory?

**Comment by theseanything** [comment]
Ah right, thanks for pointing that out! In which case we may make that the default behaviour or provide it via a flag.

**Comment by b-t-g** [comment]
Fixed by #189.
